app/background/wallet/service.js

```diff
@@ -269,42 +328,87 @@ class WalletService {
   };
 
   zap = async () => {
-    this._ensureClient();
+    await this._ensureClient();
     return this.client.zap(WALLET_ID, 'default', 1);
   };
 
   importName = (name, start) => {
     return this._executeRPC('importname', [name, start]);
   };
 
-  rpcGetWalletInfo = () => {
-    return this._executeRPC('getwalletinfo', []);
+  rpcGetWalletInfo = async () => {
+    return await this._executeRPC('getwalletinfo', []);
   };
 
   _onNodeStart = async (networkName, network, apiKey) => {
+    const conn = await getConnection();
+
     this.networkName = networkName;
+    this.apiKey = apiKey;
+    this.network = network;
+
     const walletOptions = {
       network: network,
-      port: network.walletPort,
-      apiKey,
+      port: 12137,
+      apiKey: crypto.randomBytes(20).toString('hex'),
```

Comment:
This api key is not displayed to the user anywhere? In the settings tab, the API KEY that can be revealed is probably for the full node, not the wallet (which I guess now has a different key?) This again makes using CLI tools with Bob very difficult